include/utils.hpp

```diff
@@ -49,4 +88,35 @@ class Singleton : public NonCopyable {
 template<class T>
 T* Singleton<T>::instance_ = nullptr;
 
+class OutFileStream : public NonCopyable {
+public:
+    OutFileStream() = default;
+    OutFileStream(const fs::path& path) { this->set_path(path); }
+    ~OutFileStream() { if (this->is_open()) this->close(); }
```

Comment:
Check, that `ofstream` will be closed in the its own destructor. If yes, then we do not need to close it manually.

Reply:
double check, is stream opened or not.

Reply:
> Check, that `ofstream` will be closed in the its own destructor. If yes, then we do not need to close it manually.

Alright. Fixed.